gjkr/message.go

```diff
@@ -0,0 +1,26 @@
+package gjkr
+
+import "threshold.network/roast/ephemeral"
+
+// ephemeralPublicKeyMessage is a message payload that carries the sender's
+// ephemeral public keys generated for all other group members.
+//
+// The receiver performs ECDH on a sender's ephemeral public key intended for
+// the receiver and on the receiver's private ephemeral key, creating a symmetric
+// key used for encrypting a conversation between the sender and the receiver.
+// In case of an accusation for malicious behavior, the accusing party reveals
+// its private ephemeral key so that all the other group members can resolve the
+// accusation looking at messages exchanged between accuser and accused party.
+// To validate correctness of accuser's private ephemeral key, all group members
+// must know its ephemeral public key prior to exchanging any messages. Hence,
+// this message contains all the generated public keys and it is broadcast
+// within the group.
+type ephemeralPublicKeyMessage struct {
+	senderIndex memberIndex // i
+
+	ephemeralPublicKeys map[memberIndex]*ephemeral.PublicKey // j -> Y_ij
+}
+
+func (m *ephemeralPublicKeyMessage) senderIdx() memberIndex {
```

Comment:
I think `Idx` can be easily confused with `Id`. What about the full name:
```suggestion
func (m *ephemeralPublicKeyMessage) senderIndex() memberIndex {
```

Reply:
This is what I started with but the compiler does not like the fact the field name and the function name are the same. I would be happy to remove this function altogether but we need a type constraint for `func deduplicateBySender[T interface{ senderIndex() memberIndex }](list []T,) []T`. Since `senderIdx` is used only there, I figured this might be okay.

Reply:
I refactored it in 02c828c3c1e7507bfa37f7690ea2a3004c565a3d. After adding `sessionID`, this conflict was not the only one so I ended up with `get*` functions.